### PR TITLE
Fix meta key+click does not open link in new tab

### LINKED ISSUE #8: Link clicks do not open in new tab
Trying to CMD + click on link within element does not open in new tab, takes over current tab.

I see that you are hijacking links to make the scrolling work.

### PR DESCRIPTION
**Pull request by ernsheong**
Fixes #8 

I manually tested on the demo/index.html page, and it works! Anchor scroll still works.
Didn't invest time to write the unit test, sorry :/

### CODE REVIEW

File: src/zero-md.js [hunk 1]
```diff
@@ -165,12 +165,11 @@ class ZeroMd extends HTMLElement {
...
-      window.open(window.location.href  + path[0].hash, '_blank');
-    } else {
+
+    const link = path[0];
+    if (link.hash && link.href === window.location.href) {
       this._scrollTo(path[0].hash);
-      window.location = path[0].href;
+      ev.preventDefault();
     }
   }
```
Comment (zerodevx) [inline src/zero-md.js]: Thanks for the quick work! It suddenly occurred to be that we should still check for CMD+click for anchor-link-scroll cases. The expected result if a user CMD+clicks on a hash link is that a new tab opens with the page duplicated and scrolled to the correct heading. 

Agree that we don't have to `preventDefault` none hash-link cases. 

Another thought is the check `link.hash && link.href === window.location.href` might not correctly isolate out valid hash-link cases. Take for example if a user's browser is at "https://example.com/page", and the link href is "https://example.com/page#section1"...

Perhaps this function can be refactored to:

```js
  _hijackLinks(ev) {
    let path = ev.path || ev.composedPath();
    if (path[0].tagName !== 'A') { return; }
    // check that it's a hash-link case
    if (path[0].hash && (path[0].origin + path[0].pathname === window.location.origin + window.location.pathname)) {
      ev.preventDefault();
      if (ev.metaKey) {
        window.open(window.location.href  + path[0].hash, '_blank');
      } else
        this._scrollTo(path[0].hash);
        window.location = path[0].href;
      }
    }
  }
```

Comment (ernsheong) [reply src/zero-md.js]: Ok good point, I will refactor

**Comment by ernsheong** [comment]
I have updated the PR. Not familiar with Github Review feature, might have missed some workflow step.

**Comment by zerodevx** [comment]
LGTM, thank you for your time! 👍

**Comment by ernsheong** [comment]
Sweet, thanks 👍

**Pull request merged**